frontend/pages/DashboardPage/cards/MDM/MDMSolutionsTableConfig.tsx

```diff
@@ -61,27 +57,17 @@ export const generateSolutionsTableHeaders = (): IDataColumn[] => [
   },
 ];
 
-const enhanceSolutionsData = (
-  solutions: IMdmSolution[],
-  selectedPlatformLabelId?: number
-): IMdmSolution[] => {
-  return Object.values(solutions).map((solution) => {
-    return {
-      id: solution.id,
-      name: solution.name || "Unknown",
-      server_url: solution.server_url,
-      hosts_count: solution.hosts_count,
-      selectedPlatformLabelId,
-    };
-  });
-};
-
 export const generateSolutionsDataSet = (
-  solutions: IMdmSolution[] | null,
-  selectedPlatformLabelId?: number
+  solutions: IMdmSolution[] | null
 ): IMdmSolution[] => {
   if (!solutions) {
     return [];
   }
-  return [...enhanceSolutionsData(solutions, selectedPlatformLabelId)];
+
+  return solutions.map((solution) => {
+    return {
+      ...solution,
+      displayName: solution.name ?? "Unknown",
```

Comment:
Isn't false-y name being addressed in the new `reduceSolutionsToObj` function such that we don't need thie function any more?

Reply:
This is still needed. The reduce method only organises the MDM solutions data so that we can generate a map of solutions based on their name. Since the `name` for a solution is `string | null` we need to differentiate between `null`, `""`, and normal string.

I had to update a few more things to make this differentiation, but ultimately I cannot override the `name` attribute as it can mean different mdm solutions for different values and I wouldn't want to roll up `null` and `""`. so instead I keep the name the same and add a display name for the table.